host_access/sycl-1.2.1/host_access.md

```diff
@@ -0,0 +1,130 @@
+# Temporary buffer
+
+| Proposal ID | CP017 |
+|-------------|--------|
+| Name | Host access |
+| Date of Creation | 17 September 2017 |
+| Target | SYCL 1.2.1 vendor extension |
+| Current Status | _Work In Progress_ |
+| Reply-to | Ruyman Reyes <ruyman@codeplay.com> |
+| Original author | Ruyman Reyes <ruyman@codeplay.com> |
+
+## Overview
+
+Developers in some situations want to create memory that can only be accessed from the device. This can enable the underlying device driver to allocate memory on places where it is more efficient for the device to access than the host. 
+This is possible in OpenCL 1.2 by using the CL_MEM_HOST_NO_ACCESS, CL_MEM_HOST_READ_ONLY or CL_MEM_HOST_WRITE_ONLY flags during buffer creation.
+
+A SYCL implementation for OpenCL needs to analyze the access pattern at compile time, or replace the underlying cl_mem object, in order to infer the host access pattern and create buffers with the appropriate flags.
+
+This proposal enables developers to specify the intended use pattern of a buffer, facilitating a SYCL runtime the usage of optimized buffer flags creation.
+
+## Revisions
+
+This is the first revision of this proposal.
+
+## Interface changes
+
+This proposal adds a new property for buffers, *buffer::property::host_access*.
+A SYCL buffer with the *host_access* property can only be used within one context,
+and its values are only accessible from queues that are associated with
+said context.
```

Comment:
What would happen if the bound context is a host mode context, would it be possible to create a buffer that's not accessible from anywhere?